timely_beliefs/beliefs/classes.py

```diff
@@ -1,46 +1,49 @@
 from typing import List
 from datetime import datetime, timedelta
 
-from pandas import DataFrame, DatetimeIndex, MultiIndex, TimedeltaIndex
+import pandas as pd
 from sqlalchemy import Column, DateTime, Integer, Interval, Float, ForeignKey
-from sqlalchemy.ext.declarative import declared_attr
 from sqlalchemy.ext.hybrid import hybrid_method, hybrid_property
 from sqlalchemy.orm import relationship, backref
 
-from base import Base, session
+from timely_beliefs.base import Base, session
 from timely_beliefs import Sensor
 from timely_beliefs import BeliefSource
+from timely_beliefs.sensors.classes import DBSensor
 from timely_beliefs.utils import enforce_utc
 from timely_beliefs.sensors import utils as sensor_utils
 from timely_beliefs.beliefs import utils as belief_utils
 
 
-class TimedBelief(Base):
+class TimedBelief(object):
     """"""
 
-    @declared_attr
-    def __tablename__(cls):
-        return cls.__name__.lower()
+    event_start: datetime
+    belief_horizon: timedelta
+    event_value: float
+    sensor: Sensor
+    source: str  # TODO: needs to be an object as well?
```

Comment:
You do initialize with the object, but convert to a string? If you really need to convert, why not store the source_id (i.e. an integer) similar to what we do in the BeliefsDataFrame? The init() function of the BeliefsDataFrame will look for the source_id attribute in any case.